### PR TITLE
{Network} application gateway 100% support

### PR DESCRIPTION
**Pull request by kairu-ms**
**Description**

**Testing Guide**

**History Notes**

[Network] `az network application-gateway show-backend-health`: support probe operation arguments.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
network

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 1]
```diff
@@ -670,6 +670,70 @@ def remove_trusted_client_certificate(cmd, resource_group_name, application_gate
...
 
+def show_ag_backend_health(cmd, client, resource_group_name, application_gateway_name, expand=None,
+                           protocol=None, host=None, path=None, timeout=None, host_name_from_http_settings=None,
+                           match_body=None, match_status_codes=None, address_pool=None, http_settings=None):
+    from azure.cli.core.commands import LongRunningOperation
+    if cmd.supported_api_version(min_api='2019-04-01'):
+        SubResource, ApplicationGatewayOnDemandProbe, ApplicationGatewayProbeHealthResponseMatch = cmd.get_models(
+            "SubResource", "ApplicationGatewayOnDemandProbe", "ApplicationGatewayProbeHealthResponseMatch")
+        probe_request = ApplicationGatewayOnDemandProbe()
+        probe_assigned = False
```
Comment (msyyc) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:681]: Maybe there code could be more concise?

File: src/azure-cli/azure/cli/command_modules/network/commands.py [hunk 2]
```diff
@@ -477,7 +477,7 @@ def load_command_table(self, _):
         g.custom_command('list', 'list_application_gateways')
         g.command('start', 'begin_start')
         g.command('stop', 'begin_stop')
-        g.command('show-backend-health', 'begin_backend_health', min_api='2016-09-01')
+        g.custom_command('show-backend-health', 'show_ag_backend_health', min_api='2016-09-01', client_factory=cf_application_gateways)
```
Comment (msyyc) [inline src/azure-cli/azure/cli/command_modules/network/commands.py:480]: will it cause breaking change?

Comment (kairu-ms) [reply src/azure-cli/azure/cli/command_modules/network/commands.py:480]: No

**Review by msyyc — APPROVED** [review]

**Pull request merged**